index.js

```diff
@@ -14,34 +24,82 @@ var allowCrossDomain = function (req, res, next) {
+    const splitSlug = urlString => {
+        const parsedUrl = url.parse(urlString);
+        const pathSegments = parsedUrl.pathname.split('/');
+        return pathSegments.at(-1);
+    };
+
+    const logData = {
+        buildings: result.buildings.map(item => splitSlug(item.url)),
+        activities: result.activities.map(item => splitSlug(item.url)),
+        organizations: result.organizations.map(item => splitSlug(item.url)),
+        people: result.people.map(item => splitSlug(item.url)),
+    };
+
+    const { hostname, protocol, port: urlPort, path: urlPath } = url.parse(endpointUrl);
+
+    const port = urlPort || (protocol === 'https:' ? 443 : 80);
+    const path = `${urlPath}?api_key=${apiKey}`;
+    const jsonLogData = JSON.stringify(logData);
+    const httpLib = protocol === 'https' ? https : http;
 
-var aws = require('aws-sdk');
+    const options = {
+        hostname, port, path,
+        method: 'POST',
+        headers: {
+            'Content-Type': 'application/json',
+            'Content-Length': Buffer.byteLength(jsonLogData)
+        }
+    };
+
+    const req = httpLib.request(options);
+    req.write(jsonLogData);
+
+    req.on('error', (error) => {
+      if (error.code === 'ENOTFOUND') {
+        console.error(`Request error: Invalid hostname: ${options.hostname}`);
+      } else if (error.code === 'ECONNREFUSED') {
+        console.error(`Request error: Connection refused: ${options.hostname}:${options.port}`);
+      } else if (error.code === 'ECONNRESET') {
+        console.error(`Request error: Connection reset: ${options.hostname}:${options.port}`);
+      } else if (error.code === 'ETIMEDOUT') {
+        console.error(`Request error: Connection timed out: ${options.hostname}:${options.port}`);
+      } else {
+        console.error(`Request error: ${error}`);
+      }
+    });
+
+    req.end();
+};
 
+const app = express();
+app.use(allowCrossDomain);
 app.set('port', (process.env.PORT || 5000));
 
 // this should just return a status true or something
-app.get('/', function (request, response) {
+app.get('/', (request, response) => {
     response.json({status: true, code: 'realgraph-listen'});
 });
 
-app.get('/realgraph/listen', function (request, response) {
+app.get('/realgraph/listen', (request, response) => {
     if (!request.query.url.startsWith(process.env.ALLOWED_ADDRESS_PATTERN)) {
         response.json({status: false, message: 'Address not allowed'});
         return;
     }
 
-    var urlPath = getPathFromUrl(request.query.url);
-    var queryString = "INSERT INTO realgraph_pings(url, hash) VALUES ($1, md5($2))";
-    pg.connect(process.env.DATABASE_URL, function (err, client, done) {
-        client.query(queryString, [urlPath, urlPath], function (err, result) {
+    const urlPath = getPathFromUrl(request.query.url);
+    const queryString = "INSERT INTO realgraph_pings(url, hash) VALUES ($1, md5($2))";
+    pg.connect(process.env.DATABASE_URL, (err, client, done) => {
```

Comment:
I would instantiate this in separate module to have singleton. Now new connection to database is made on every endpoint hit. Singleton for pg (and dynamo) will automatically handle connections or connection pools.